sdk/samples/iot/paho_iot_hub_pnp_component_example.c

```diff
@@ -0,0 +1,870 @@
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      &options));
+
+  return AZ_OK;
+}
+
+static void send_device_info(void)
+{
+  // Get the device info in a JSON payload and the topic to which to send it
+  az_result result;
+  if ((result = sample_pnp_device_info_get_report_data(&client, &publish_message)) != AZ_OK)
+  {
+    LOG_ERROR("Could not get the device info data: error code = 0x%08x", result);
+    exit(result);
+  }
+
+  // Send the MQTT message to the endpoint
+  mqtt_publish_message(publish_message.topic, publish_message.out_payload_span, SAMPLE_PUBLISH_QOS);
+
+  // Receive response for device info publish
+  mqtt_receive_message();
+}
+
+// Callback invoked by pnp functions each time it finds a property in the twin document
+static void sample_property_callback(
+    az_span component_name,
+    az_span property_name,
+    az_json_token* property_value,
+    int32_t version,
+    void* user_context_callback)
+{
+  (void)user_context_callback;
+  if (az_span_ptr(component_name) == NULL || az_span_size(component_name) == 0)
+  {
+    LOG_ERROR(
+        "Property=%.*s arrived for Control component itself. This does not support\
+                writable properties on it (all properties are on sub-components)",
+        az_span_size(property_name),
+        az_span_ptr(property_name));
+  }
+  else if (
+      sample_pnp_thermostat_process_property_update(
+          &client,
+          &sample_thermostat_1,
+          component_name,
+          property_name,
+          property_value,
+          version,
+          &publish_message)
+      == AZ_OK)
+  {
+    LOG_SUCCESS("Updated property on thermostat 1");
+
+    // Send response to the updated property
+    mqtt_publish_message(
+        publish_message.topic, publish_message.out_payload_span, SAMPLE_PUBLISH_QOS);
+  }
+  else if (
+      sample_pnp_thermostat_process_property_update(
+          &client,
+          &sample_thermostat_2,
+          component_name,
+          property_name,
+          property_value,
+          version,
+          &publish_message)
+      == AZ_OK)
+  {
+    LOG_SUCCESS("Updated property on thermostat 2");
+
+    // Send response to the updated property
+    mqtt_publish_message(
+        publish_message.topic, publish_message.out_payload_span, SAMPLE_PUBLISH_QOS);
+  }
+  else
+  {
+    LOG_ERROR("There was an error updating a property");
+  }
+}
+
+// Switch on the type of twin message and handle accordingly
+static void handle_twin_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_twin_response* twin_response)
+{
+  az_result result;
+
+  az_json_reader json_reader;
+  az_span twin_payload_span;
+
+  if (message->payloadlen)
+  {
+    LOG_SUCCESS("Payload: %.*s", message->payloadlen, (char*)message->payload);
+    twin_payload_span = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+    if ((result = az_json_reader_init(&json_reader, twin_payload_span, NULL)) != AZ_OK)
```

Comment:
Same here/elsewhere. Don't do direct comparison to a particular `az_result` if all you care about is whether the operation succeeded or failed.